### PR TITLE
feat: create the CapTP connection facet

### PR DESCRIPTION
**Pull request by michaelfig**
Steps toward using CapTP in the demo.

Also, some cleanups to reenable the use of encouragement by nickname.

### CODE REVIEW

**Comment by michaelfig** [comment]
Please review with "hide whitespace changes".

File: contract/src/contract.js [hunk 1]
```diff
@@ -77,17 +87,17 @@ const start = async zcf => {
...
     count += 1;
     updateNotification();
-    return encouragement;
+    return `${personalize(nickname)}${encouragement}`;
   };
 
   const publicFacet = {
-    makeInvitation() {
-      return zcf.makeInvitation(encourage, 'encouragement');
+    makeInvitation(nickname = undefined) {
```
Comment (katelynsills) [inline contract/src/contract.js:94]: hmm, I think we could simplify this by using a default nickname like `friend` and removing the `personalize` function entirely. The way `personalize` gets called is kind of hard to read, and I think we want to make this app as simple as possible.

File: api/src/server.js [hunk 2]
```diff
@@ -163,7 +163,7 @@ export const bootPlugin = ({ getState, setState }) => {
             case 'encouragement/getEncouragement': {
               send({
                 type: 'encouragement/getEncouragementResponse',
-                data: await E(publicFacet).getFreeEncouragement(),
+                data: await E(publicFacet).getFreeEncouragement(obj.nickname),
```
Comment (katelynsills) [inline api/src/server.js:166]:
```suggestion
                data: await E(publicFacet).getFreeEncouragement(obj.data.nickname),
```
I could misunderstanding, but I think this should have the same data structure as `encouragement/sendInvitation`

Comment (katelynsills) [reply api/src/server.js:166]: Just tested locally, and the original seems to work. I think users will find the different confusing, or at least I do :)

**Review by katelynsills — APPROVED** [review]
Looks good! I think we can make some things simpler and more uniform.

File: api/src/server.js (hunk 2 above)
Comment (michaelfig) [reply api/src/server.js]: > I could misunderstanding, but I think this should have the same data structure as `encouragement/sendInvitation`

Good catch!  I fixed this.

File: contract/src/contract.js (hunk 1 above)
Comment (michaelfig) [reply contract/src/contract.js:90]: Done.

**Pull request merged**